src/utils/Subscription.js

```diff
@@ -25,15 +23,13 @@ function initListeners() {
       let isSubscribed = true
       if (next === current) next = current.slice()
       next.push(listener)
-      count++
 
       return function unsubscribe() {
-        if (!isSubscribed || count === 0) return
+        if (!isSubscribed || !current) return
```

Comment:
Hmm, I wonder if that should reset back to an empty array then? nulls are potentially dangerous when they sneak into the code to actually mean a 3rd (or sometimes 4th) state. It usually means you need another flag to make that state more explicit, which you already have with `isSubscribed`. I've got a whiff of a code smell here...